lib/ExtendTruffleContract.js

```diff
@@ -0,0 +1,122 @@
+import { Utils } from "./utils";
+import { config } from "./config.js";
+/**
+ * Abstract base class for all Arc contract wrapper classes
+ *
+ * Example of how to define a sub class:
+ *
+ *  export class AbsoluteVote extends ExtendTruffleContract(SolidityAbsoluteVote) {}
+ *
+ * @param {TruffleContract} superclass
+ */
+export const ExtendTruffleContract = superclass =>
+  class {
+    constructor(contract) {
+      // Make sure this contract is configured with the web3 provider and default config values
+      superclass.setProvider(Utils.getWeb3().currentProvider);
+      superclass.defaults({
+        from: Utils.getDefaultAccount(), // Use web3.eth.defaultAccount as the from account for all transactions
+        gas: config.get("gasLimit")
+      });
+      this.contract = contract;
+      for (const i in this.contract) {
+        if (this[i] === undefined) {
+          this[i] = this.contract[i];
+        }
+      }
+    }
+
+    static async new() {
+      return superclass.new().then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
+
+    static async at(address) {
+      return superclass.at(address).then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
+
+    static async deployed() {
+      return superclass.deployed().then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
+
+    /**
+     * Call setParameters on this scheme.
+     * Returns promise of ArcTransactionResultReturnType where Result is the parameters hash.
+     * If there are any parameters, then this function must be overridden by the subclass to provide them.
+     *
+     * @param {any} params -- object with properties whose names are expected by the scheme to correspond to parameters.
+     * Currently all params are required, contract wrappers do not as yet apply default values.
+     */
+    async setParams(params) {
```

Comment:
looking at how _setParameters is used below why not just have setParams do what _setParameters does and use the spread operator to pass along all params, then in most cases the subclass can just use the parent version of setParams

Reply:
I no longer recall why I did it that way, so I'm reverting to your suggestion.